### PR TITLE
Fix: `no-dupe-keys` false negative (fixes #6801)

### LINKED ISSUE #6801: no-dupe-keys: false negative between a normal property and a getter.
**What version of ESLint are you using?**

- 3.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

    $ echo "var obj = {a:1, get a() { return 1 }}" | eslint --stdin --rule no-dupe-keys:2 --env es6 --no-eslintrc

**What did you expect to happen?**

- Duplicate key 'a'. (no-dupe-keys)

**What actually happened? Please include the actual, raw output from ESLint.**

- No errors.

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #6801.

This PR fixes a false negative between a normal property and a getter/setter.
In addition, I upgrade code by ES2015.

### CODE REVIEW

File: lib/rules/no-dupe-keys.js [hunk 1]
```diff
@@ -6,6 +6,79 @@
...
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const GET_KIND = /^(?:init|get)$/;
```
Comment (ilyavolodin) [inline lib/rules/no-dupe-keys.js]: Sorry, I'm not sure what the first part of this regEx is supposed to check?

Comment (mysticatea) [inline lib/rules/no-dupe-keys.js]: Normal properties' `kind` is `"init"`. This regexp is to address normal properties as getter/setter pair. As a result, this rule will address `{a: 1, get a() {}}` as duplicate keys. This is the purpose of this PR :)

Comment (ilyavolodin) [inline lib/rules/no-dupe-keys.js]: Ah, OK, now I understand, thanks.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**